pygmt/helpers/utils.py

```diff
@@ -220,6 +221,9 @@ def launch_external_viewer(fname):
         os.startfile(fname)  # pylint: disable=no-member
     else:
         webbrowser.open_new_tab(f"file://{fname}")
+    # suspend the execution for 0.5 s to avoid the images being deleted
+    # when a Python script exits
+    time.sleep(0.5)
```

Comment:
Instead of using `time.sleep`, could we change the `subprocess.run` to be blocking? I.e. ensure that the subprocess run is completed before continuing to exit the function.

Reply:
I tried a few ways but still can't make it blocking.

Reply:
Do you think time.sleep is better than using subprocess.Popen(...).wait()?

Reply:
> subprocess.Popen(...).wait()

Unfortunately, changing `subprocess.call` to `subprocess.Popen(...).wait()` doesn't work at all.